src/core/tritonserver.h

```diff
@@ -667,6 +667,16 @@ TRITONSERVER_InferenceRequestOutputShape(
     TRITONSERVER_InferenceRequest* inference_request, const char* name,
     const int64_t** shape, uint64_t* dim_count);
 
+/// Get the names of all output tensors.
+/// \param inference_request The request object.
+/// \param names The names of the outputs.
```

Comment:
Specify that the names is owned by the function caller and must be pre-allocated with enough space.
However, that is unsafe, for instance, the HTTP frontend implementation below assumes output counts <= 10 and segfault will happen otherwise. Should either let infer request object own the names, or create it in place and let function caller to release it on finish.